src/dbcp/transform/gridstatus.py

```diff
@@ -591,8 +591,28 @@ def _create_project_status_classification_from_multiple_columns(
     return iso_df
 
 
-def _transform_miso(iso_df: pd.DataFrame) -> pd.DataFrame:
-    """Make miso specific transformations."""
+def _transform_miso(post_2017: pd.DataFrame, pre_2017: pd.DataFrame) -> pd.DataFrame:
+    """Make miso specific transformations.
+
+    In the second half of 2023, MISO removed all projects that entered the queue prior to 2017.
+    Luckily, our first snapshot of MISO data from GS includes projects that entered the queue
+    prior to 2017. This function grabs all unique projects from both snapshots and combines them.
+
+    Args:
+        post_2017: MISO data from 2017 to present. This contains the latest MISO data.
+        pre_2017: The oldest snapshot of GS we have. Happens to include prior to 2017.
+    """
+    # grab projects that are only in pre_2017
+    only_in_pre_2017 = pre_2017[~pre_2017["queue_id"].isin(post_2017["queue_id"])]
+
+    # ensure there are no active projects in only_in_pre_2017
+    assert (
+        only_in_pre_2017["queue_status"].ne("Active").all()
+    ), "There are active projects in the pre-2017 MISO data that are not in the current MISO data."
+
+    # concat only_in_pre_2017 with iso_df
+    iso_df = pd.concat([post_2017, only_in_pre_2017])
```

Comment:
I saw that there are no column differences between the two datasets, but I thought there would be because of code changes in GridStatus since the old archive was created. For my own background context, why is that not the case? Did the archive happen after the GS changes? Was MISO not one of the affected ISOs? Or did something other than column names change?

Reply:
It doesn't look like there have been any column name changes in the [last couple of years](https://github.com/kmax12/gridstatus/blame/main/gridstatus/miso.py).

Reply:
I mean isn't that old archive of processed GridStatus outputs, the code for which has changed since then? Or is that archive of raw data that is safe from those changes?